From 5c6b36ff4b9c5fc82c80c62c7832e515f3bbf7de Mon Sep 17 00:00:00 2001
From: Paul Handrigan <Paul.Handrigan@cirrus.com>
Date: Fri, 22 Jan 2016 14:57:59 -0600
Subject: [PATCH 656/959] IKSWM-15398: ASoC: cs35l34: Change enum to
 SINGLE_DECL

Change kcontrol enum to SINGLE_DECL to simplify the code.

Change-Id: I1dae2195b247aa6e058abf7256237f43568f6f1c
Signed-off-by: Paul Handrigan <Paul.Handrigan@cirrus.com>
Reviewed-on: http://gerrit.mot.com/816721
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 sound/soc/codecs/cs35l34.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/sound/soc/codecs/cs35l34.c b/sound/soc/codecs/cs35l34.c
index e9250ee4779..e5e60ef1747 100644
--- a/sound/soc/codecs/cs35l34.c
+++ b/sound/soc/codecs/cs35l34.c
@@ -298,17 +298,12 @@ static int cs35l34_main_amp_event(struct snd_soc_dapm_widget *w,
 
 static DECLARE_TLV_DB_SCALE(dig_vol_tlv, -10200, 50, 0);
 
-static const int gain_ranges[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0xA, 0xB,
-				0xC, 0xD, 0xE, 0xF};
-
 static const char * const gain_labels[] = {"3dB", "4dB", "5dB", "6dB", "7dB",
 			"8dB", "9dB", "10dB", "11dB", "12dB", "13dB", "14dB",
 			"15dB", "16dB", "17dB", "18dB"};
 
-static const struct soc_enum amp_gain = SOC_VALUE_ENUM_SINGLE(
-				CS35L34_AMP_ANLG_GAIN_CTL,
-				0, 0xF, ARRAY_SIZE(gain_labels),
-				gain_labels, gain_ranges);
+static const SOC_ENUM_SINGLE_DECL(amp_gain, CS35L34_AMP_ANLG_GAIN_CTL, 0,
+	gain_labels);
 
 static const struct snd_kcontrol_new cs35l34_snd_controls[] = {
 	SOC_SINGLE_SX_TLV("Digital Volume", CS35L34_AMP_DIG_VOL,
-- 
2.11.0

